machine_learning/data_downloader.py

```diff
@@ -0,0 +1,147 @@
+
+    Parameters
+    ----------
+    url : string
+
+    Returns
+    -------
+    from_version_id, to_version_id : string
+        To and from version ids extracted from the url
+    """
+    result = urlparse(url)
+    if (result.netloc == 'versionista.com'):
+        ids = [element for element in result.path.split('/') if element != '']
+        version_ids = ids[-1].split(':')
+        to_version_id = version_ids[0]
+        from_version_id = version_ids[1]
+        return from_version_id, to_version_id
+    else:
+        return None, None
+
+def get_storage_uris(url):
+    """
+    Get uris of the versions stored in our db
+    Uses get_version_uri from web_monitoring.db
+
+    Parameters
+    ----------
+    url : string
+
+    Returns
+    -------
+    from_version_uri, to_version_uri : string
+        To and from version uris to access content in db
+    """
+    from_version_id, to_version_id = parse_url(url)
+
+    if (from_version_id == None and to_version_id == None):
+        return None, None
+    else:
+        if from_version_id == '0':
+            to_version_uri, valid_from_version_id = get_version_uri(version_id=to_version_id,
+                                                                    id_type='source',
+                                                                    source_type='versionista',
+                                                                    get_previous=True)
+            from_version_uri = get_version_uri(version_id=valid_from_version_id,
+                                               id_type='source',
+                                               source_type='versionista')
+        else :
+            to_version_uri = get_version_uri(version_id=to_version_id,
+                                             id_type='source',
+                                             source_type='versionista')
+            from_version_uri = get_version_uri(version_id=from_version_id,
+                                               id_type='source',
+                                               source_type='versionista')
+        return from_version_uri, to_version_uri
+
+def download(filename, dirname, path=''):
+    """
+    Downloader to read csv and download content for the dataset
+
+    Parameters
+    ----------
+    filename* : string
+        The csv file which has the urls in the 'Last Two - Side by Side' column
+    dirname* : string
+        The directory name in which you want to download the content, creates a new directory
+    path : string (default : current working directory)
+        Path to the directory where the csv is stored and downloaded content will be stored
+    """
+    if (path == ''):
+        path = os.getcwd()
+    os.mkdir(dirname)
+    new_dir = os.path.join(path, dirname)
+    download_log_handler = open(str(dirname)+ '.txt', 'w')
+    filename = os.path.join(os.getcwd(), filename)
+    os.chdir(new_dir)
```

Comment:
The success of this operation should probably be checked, because if the `chdir` doesn't take place, subsequent operations will not write files to the expected locations.